### PR TITLE
feat: AddChildren Action

### LINKED ISSUE #272: EPIC: Replace LazyComponent with AddChildren action
**Describe the problem**
`LazyComponent` theoretically is no longer necessary with #54, specifically [here](https://github.com/ZupIT/beagle-web-core/tree/specification/component-interaction#addchildren). Most, use cases could be replaced with non trivial uses of the context and actions.

**Expected behavior**
An action whose behavior is replacing a component completely in a component tree.

### PR DESCRIPTION
**Pull request by dantes-git**
## Description

Implements [AddChildren](https://github.com/ZupIT/beagle-web-core/tree/specification/component-interaction#addchildren) Action

## Related Issues

#272 

## Tests

I added the following tests:

testAddChildrenDefault
testAddChildrenPrepend
testAddChildrenReplace

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [DCO].
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

[issue database]: https://github.com/ZupIT/beagle/issues
[DCO]: https://developercertificate.org/

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: iOS/Example/BeagleDemo/BeagleDemo/Screens/AddChildrenScreen.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                    text: "Add Children",
+                    onPress: [
+                        AddChildren(componentId: "containerId", value: [ Text("@{context}")], mode: .prepend)
+                    ]
+                )
+            }
+        }
+    }
+}()
+
```
Comment (botbeaglezup) [inline iOS/Example/BeagleDemo/BeagleDemo/Screens/AddChildrenScreen.swift:42]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/846c371cd7a7b04818367187ad40f48fe6e2f7af/iOS/Example/BeagleDemo/BeagleDemo/Screens/AddChildrenScreen.swift#L42"></span>Files should have a single trailing newline.<br />
<code>trailing_newline</code> <code>AddChildrenScreen.swift:42</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Sources/Beagle/Sources/Action/Types/Tests/AddChildrenTests.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        // Given
+        let addChildren = AddChildren(componentId: "", value: [Text("text 1"), Text("text 2")])
+        let view = UIView()
+        let controller = BeagleControllerNavigationSpy()
+
+        // When
+        addChildren.execute(controller: controller, origin: view)
+
+        // Then
+//        TODO: take snapshot
```
Comment (botbeaglezup) [inline iOS/Sources/Beagle/Sources/Action/Types/Tests/AddChildrenTests.swift:34]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/846c371cd7a7b04818367187ad40f48fe6e2f7af/iOS/Sources/Beagle/Sources/Action/Types/Tests/AddChildrenTests.swift#L34"></span>TODOs should be resolved (take snapshot).<br />
<code>todo</code> <code>AddChildrenTests.swift:34</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Sources/Beagle/Sources/Action/Types/Tests/AddChildrenTests.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        // Given
+        let addChildren = AddChildren(componentId: "", value: [Text("text")], mode: .prepend)
+        let view = UIView()
+        let controller = BeagleControllerNavigationSpy()
+
+        // When
+        addChildren.execute(controller: controller, origin: view)
+
+        // Then
+//        TODO: take snapshot
```
Comment (botbeaglezup) [inline iOS/Sources/Beagle/Sources/Action/Types/Tests/AddChildrenTests.swift:47]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/846c371cd7a7b04818367187ad40f48fe6e2f7af/iOS/Sources/Beagle/Sources/Action/Types/Tests/AddChildrenTests.swift#L47"></span>TODOs should be resolved (take snapshot).<br />
<code>todo</code> <code>AddChildrenTests.swift:47</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Sources/Beagle/Sources/Action/Types/Tests/AddChildrenTests.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        // Given
+        let addChildren = AddChildren(componentId: "", value: [Text("text")], mode: .replace)
+        let view = UIView()
+        let controller = BeagleControllerNavigationSpy()
+
+        // When
+        addChildren.execute(controller: controller, origin: view)
+
+        // Then
+//        TODO: take snapshot
```
Comment (botbeaglezup) [inline iOS/Sources/Beagle/Sources/Action/Types/Tests/AddChildrenTests.swift:60]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/846c371cd7a7b04818367187ad40f48fe6e2f7af/iOS/Sources/Beagle/Sources/Action/Types/Tests/AddChildrenTests.swift#L60"></span>TODOs should be resolved (take snapshot).<br />
<code>todo</code> <code>AddChildrenTests.swift:60</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Sources/Beagle/Sources/Action/Types/AddChildren.swift [hunk 5]
```diff
@@ -22,18 +22,21 @@ extension AddChildren: Action {
...
-            break
+            views.forEach { view.addSubview($0) }
         case .prepend:
-            views = views + view.subviews
-            view.subviews.forEach { $0.removeFromSuperview() }
+            var index = 0
+            for subView in views {
+                view.insertSubview(subView, at: index)
+                index += 1
+            }
```
Comment (theffc) [inline iOS/Sources/Beagle/Sources/Action/Types/AddChildren.swift:35]:
```suggestion
            for (index, subView) in views.enumerated() {
                view.insertSubview(subView, at: index)
            }
```

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**